### ISSUE TITLE
[Question]Generating video file took few seconds and it's prolong significantly the test duration

### ISSUE THREAD

**Issue by hananmalka** [microsoft/playwright#5350]
Hi, 
I have a suite of tests that takes ~3 minutes **without** using the awesome video feature and ~4+ minutes (133%) **with** the video feature. 

I looked at the logs and saw that each context closing takes few more seconds than without video. 

```
2021-02-08T10:52:04.726Z pw:api => browserContext.close started
2021-02-08T10:52:15.962Z pw:api <= browserContext.close succeeded
Success notification after 11 seconds
```
```
2021-02-08T10:51:11.580Z pw:api => browserContext.close started
2021-02-08T10:51:20.406Z pw:api <= browserContext.close succeeded
Success notification after 9 seconds
```

Is there a way to prevent this duration increase?

**Comment by mxschmitt** [comment]
Could you provide more details about your environment? Playwright version, browser, Operating system.

(The Chromium implementation afaik is processing it fully at the end so its depended on how long your page instance was opened how long the close and by that the video processing takes.)

**Comment by hananmalka** [comment]
Thanks @mxschmitt for the response.
Details:
Playwright version: 1.8.0
Chromium version:  90.0.4392.0
OS: MacOS Catalina (10.15.7)

I'm creating page and context objects per test and closing them in the afterEach (jest). 
Tests duration are between 5-30 seconds.

**Comment by dgozman** [comment]
Duplicate of #5051.

**Comment by hananmalka** [comment]
> Duplicate of #5051.

@dgozman - Actually is not the same.
According to @pavelfeldman in the mentioned ticket, the reason for this is that he isn't closing the context per test. 
I **do** close the context & page after each test, and still, the notification of context close success is being reached only after 15-20 seconds.

**Comment by dgozman** [comment]
@hananmalka The actual reason is that saving video is sometimes slow, as you say in this issue. We are working on fixing this. It just happened that #5051 has an additional problem of reusing context.
